### PR TITLE
Add schemas and base related to policy manager

### LINKED ISSUE #693: Restructure the taxonomy definitions
Restructure Taxonomy json from the current format to have base + customization format. The goal is to have a single source of truth (JSON schema) suitable for all purposes (validation in manager, code generation for manager and for connectors, possibly validation in connectors, etc.) 

The new  format is:
1. `schemas`: are 1st level object definitions. Examples: catalog and policy manager request and response.
2. `base`:  are definitions that are subject to customization. Examples: base definitions for tags, actions, interfaces.
3. `layers`: are the customization over base definitions. Examples: specific tags, specific actions.

base and layers are compiled to a single `taxonomy.json` file that can be referred from the `schemas`.

Design doc: https://docs.google.com/document/d/1J7f8co97C5AXtpbgUKlSPCDMU60lkYmQnDEbx-aKuOM/

Tasks:
- [x] Tool to generate `taxonomy.json` from base and layers
- [x] Add schemas and base related to policy manager
- [ ] Add schemas and base related for everything else

### PR DESCRIPTION
**Pull request by ritwikc**
Partially Fixes #693. This PR includes:

This PR uses ideas from PR #711 and PR #637  and has the following changes:

1. Restructures the config/taxonomy folder to be conformant with base and layers architecture as proposed in [Taxonomy Technical Design.](https://docs.google.com/document/d/1J7f8co97C5AXtpbgUKlSPCDMU60lkYmQnDEbx-aKuOM/edit#)
2. The `base` folder provides schema definitions which are mandatory and generic.
3. The `layers` folder provides `customization` of the schema definitions provided by base folder.
4. An example of the customization is provided in the context of schemas needed for the policy manager in the `objects` folder.
5. Checks validation of policy manager schemas to be conformant to structural schema based Draft 4 specification of JSON      Schema standard.

### CODE REVIEW

File: config/taxonomy/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+GLANG ?= "go"
+all: clean compile gen
+compile:
+	go run ../../main.go taxonomy compile --base ./base/base.yaml --out ./objects/taxonomy.json \
+	./layers/module/actions.yaml \
+	./layers/catalog/locations.yaml \
+	./layers/catalog/tags.yaml \
+		--codegen
```
Comment (roee88) [inline config/taxonomy/Makefile:8]: For this PR no layers, just base taxonomy.

File: config/taxonomy/codegen.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+openapi: 3.0.1
```
Comment (roee88) [inline config/taxonomy/codegen.spec.yaml:1]: This file is not needed for this PR

File: config/taxonomy/layers/catalog/locations.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+definitions:
```
Comment (roee88) [inline config/taxonomy/layers/catalog/locations.yaml:1]: Should we rename the layers folder to `example` to indicate that this is an example taxonomy?

**Review by rohithdv — COMMENTED** [review]
Can change the title of the PR to "Add schemas and base related to policy manager".
In the description of the PR, you can say "Partially Fixes #693 ". Also  the following description can be added.
This PR uses ideas from [PR 711](https://github.com/mesh-for-data/mesh-for-data/pull/711) and [PR 637](https://github.com/mesh-for-data/mesh-for-data/pull/637) and has the following changes: 
1. Restructures the config/taxonomy folder to be conformant with `base` and `layers` architecture as proposed in [Taxonomy Technical Design](  https://docs.google.com/document/d/1J7f8co97C5AXtpbgUKlSPCDMU60lkYmQnDEbx-aKuOM/edit#). 
2. The `base` folder provides schema definitions which are mandatory and generic. 
3. The `layers` folder provides `customization` of the schema definitions provided by base folder. 
4. An example of the customization is provided in the context of schemas needed for the  policy manager in the `objects` folder.
5. Checks validation of policy manager schemas to be conformant to structural schema based Draft 4 specification of JSON Schema standard.

File: config/taxonomy/objects/policy_manager_request.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+$schema: http://json-schema.org/draft-04/schema#
```
Comment (roee88) [inline config/taxonomy/objects/policy_manager_request.yaml:1]: No need for both YAMLs and JSON, just keep the JSON if that's what you need.

File: pkg/text/indent.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 IBM Corp.
```
Comment (roee88) [inline pkg/text/indent.go:1]: No need for this file

File: config/taxonomy/objects/policy_manager_request.json [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		  },
+		  "processingLocation": {
+			"$ref": "taxonomy.json#/definitions/ProcessingLocation"
+		  }
+		}
+	  },
+	  "ActionType": {
+		"type": "string",
+		"enum": [
+		  "read",
```
Comment (revit13) [inline config/taxonomy/objects/policy_manager_request.json:36]: actions should also be in a layer?

Comment (roee88) [reply config/taxonomy/objects/policy_manager_request.json:36]: We do not currently see it as something configurable by any component.

File: config/taxonomy/layers/catalog/locations.yaml (hunk 3 above)
Comment (ritwikc) [reply config/taxonomy/layers/catalog/locations.yaml:1]: Yes, I think that's better.

File: config/taxonomy/Makefile (hunk 1 above)
Comment (ritwikc) [reply config/taxonomy/Makefile:8]: Done

File: config/taxonomy/codegen.spec.yaml (hunk 2 above)
Comment (ritwikc) [reply config/taxonomy/codegen.spec.yaml:1]: Done

File: config/taxonomy/layers/catalog/locations.yaml (hunk 3 above)
Comment (ritwikc) [reply config/taxonomy/layers/catalog/locations.yaml:1]: Done

File: config/taxonomy/objects/policy_manager_request.yaml (hunk 4 above)
Comment (ritwikc) [reply config/taxonomy/objects/policy_manager_request.yaml:1]: Removed YAMLs.

File: pkg/text/indent.go (hunk 5 above)
Comment (ritwikc) [reply pkg/text/indent.go:1]: Removed.

File: config/taxonomy/Makefile [hunk 7]
```diff
@@ -0,0 +1,9 @@
+GLANG ?= "go"
```
Comment (roee88) [inline config/taxonomy/Makefile:1]: remove unused

File: config/taxonomy/Makefile [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+all: clean compile gen
+compile:
```
Comment (ronenkat) [inline config/taxonomy/Makefile:3]: suggest to rename compile --> compile-taxonomy-scheme 
To differentiate from code compile

File: config/taxonomy/Makefile [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+	go run ../../main.go taxonomy compile --base ./base/base.yaml --out ./objects/taxonomy.json \
+		--codegen
```
Comment (roee88) [inline config/taxonomy/Makefile:5]: I'm not sure if we need `--codegen` here but ok for now

File: config/taxonomy/Makefile [hunk 10]
```diff
@@ -0,0 +1,9 @@
+GLANG ?= "go"
+all: clean compile gen
+compile:
+	go run ../../main.go taxonomy compile --base ./base/base.yaml --out ./objects/taxonomy.json \
```
Comment (roee88) [inline config/taxonomy/Makefile:4]: Add `ROOT_DIR := ../..` and use it instead of `../..` like in other Makefiles.

Comment (ritwikc) [reply config/taxonomy/Makefile]: Done

File: config/taxonomy/Makefile (hunk 7 above)
Comment (ritwikc) [reply config/taxonomy/Makefile]: Done

File: config/taxonomy/Makefile (hunk 8 above)
Comment (ritwikc) [reply config/taxonomy/Makefile]: Done

File: config/taxonomy/Makefile [hunk 11]
```diff
@@ -0,0 +1,10 @@
+ROOT_DIR := ../..
+all: clean compile gen
```
Comment (roee88) [inline config/taxonomy/Makefile:2]: None of these targets exist in this makefile

**Review by roee88 — CHANGES_REQUESTED** [review]

File: config/taxonomy/Makefile (hunk 11 above)
Comment (ritwikc) [reply config/taxonomy/Makefile]: Changed it.

File: config/taxonomy/objects/policy_manager_request.json (hunk 6 above)
Comment (revit13) [reply config/taxonomy/objects/policy_manager_request.json:36]: Eventually this will be used by M4Dapplication right?

**Review by roee88 — APPROVED** [review]

**Review by rohithdv — APPROVED** [review]

**Pull request merged**